### PR TITLE
Added sanitization of rendered READMEs

### PR DESCRIPTION
**Pull request by Hirevo**
This PR adds a sanitization step right after the rendering a crate's README from Markdown to HTML.  

Currently, a malicious actor can theoretically publish a crate with a README containing a `<script>` tag which would read user's cookies and send it to a server somewhere.  
This sanitization step makes sure that there is no such cross-site scripting attack possible from crate READMEs, by parsing the rendered HTML and pruning sketchy HTML tags or attributes that would allow things like code execution on the client-side.  
Basically, it makes sure the content is made of purely static HTML elements.  

This sanitization is powered by the `ammonia` crate and uses defaults settings, which seems to be sane defaults, but I think I'll have to look a bit more in detail before merging it.

### CODE REVIEW

**Comment by danieleades** [comment]
is it intended that the README rendering dependencies are included even without the `frontend` feature?

**Comment by Hirevo** [comment]
@danieleades I think this was to allow toggling the frontend on/off without having any READMEs that aren't rendered.
But the plan would most likely to, indeed, move README rendering behind the `frontend` feature and have something like a companion binary to re-render missing READMEs (or existing ones to change the syntax-highlighting theme).
It is currently listed a thing yet to do on the project's README, I just need to actually go and do it.

**Comment by Hirevo** [comment]
Ok, this has been rebased on the latest master and should be ready to merge.  
(It always has been, to be honest, sorry about that delay)
